Update iOS deployment target to 9.0 for Xcode 12

GeroHerkenrath opened the pull request:
Similar to [this PR](https://github.com/Building42/Telegraph/pull/103) in the Telegraph repo I propose incrementing the iOS deployment target to 9.0 to get rid of some annoying warnings.

I also adapted the podspec and other parts with a new version number (2.10.0) as this might be a breaking change for some people. The rest is self-explanatory I guess. 😄

GeroHerkenrath committed:
Update iOS deployment target to 9.0 for Xcode 12

JorgeFrias commented:
@GeroHerkenrath I am having problems importing this pull request with Swift Package Manager. Seems like Xcode 12 has problems fetching it, I'm unable to find the cause of the problem...

JorgeFrias commented:
@GeroHerkenrath I think I found the Issue, seems like your git version tag is not well formatted. Seems like there is an estrange character right after the version number. You can see what I'm referencing in the [Github tags](https://github.com/GeroHerkenrath/HTTPParserC/tags) for your fork note the "..." button.

GeroHerkenrath commented:
This is weird... I created the tag normally via the GitHub web UI itself. Have done so hundreds of times (in other projects) and never had an issue. The "..." usually just reveals the github specific release notes, which should not be part of the tag itself. 

Also, what do you mean with "importing this pull request with Swift Package Manager" specifically? I mean, you don't import a PR, but a specific repository's tag or a specific branch? I assume you mean either one, but could you perhaps let me know which repository you tried to import from, my fork or the original one? I guess you tried to just import from my fork, specifying the tag and SPM fails somehow. 

Unfortunately I cannot look into this right now as most of this is on my work computer and I am off today and tomorrow. I'll make myself a note, however and look into this hopefully on Tuesday. There's a few related issues with other repos I forked for this, too, I have to sort those out as well anyway. :)

Thanks for letting my know so far!

GeroHerkenrath commented:
@JorgeFrias Okay, I just checked this further and I really cannot see anything wrong. As said, the "..." on the tag is just meta information, which is empty (and probably only not completely null or something because I created it via GitHub itself). I can check out the library from my work just fine, or rather the Swift Package Manager can. I made a small test project, used my fork's HTTPS URL and defined the tag as version number (with "next major"). SPM pulled the code just fine.

Also in my work project it works, which is setup as follows: I created a branch on my fork of Telegraph in which I changed the dependencies to HTTPParserC and CocoaAsyncSocket so that they point to my forks as well (and specifying the tags in those forks for version numbers). The Xcode project then specifies this branch as dependency instead of the originals, and SPM has no problems fetching the code from there accordingly.

By the way, you can also see that the build pipeline had no problems fetching this either... I don't see what I could do?

Ultimately you can see from the changed files that I barely did anything, I just changed the deployment target and proposed a new version number (using 2.9.3 would also be fine I guess, and more in sync with how CocoaAsyncSocket did it, btw).

JorgeFrias commented:
@GeroHerkenrath  Thank you for so much consideration, I just checked with a fresh new project as you say and I have no problems pulling it with Swift Package Manager. As you say I cannot see why it didn't work, you barely did any changes so made no sense for the package manager to crash, that's why I commented, sorry for not making the reasonable thing and create a fresh test, I don't know what I was thinking...

I am very sorry, for all the trouble. I may have something broke somewhere with my main Xcode project file, or something else. I have no problems pulling your fork manually, but SPM crashes doing so only in my main project. 🙃 

Thank you.

yvbeek commented:
Thank you for the PR!

Sorry that it took me so long to have a look at this. I've moved and work has been insane. 

I've updated the http-parser code to the latest version on the https://github.com/nodejs/http-parser repo. Although I agree that the version number should probably be `2.10.0` because of the breaking change, I'd like to keep the version number aligned with that repo.

I'll close this PR because of some conflicting changes. Please let me know if I missed anything.

GeroHerkenrath commented:
Thanks for updating this! I wasn't aware this one is mirroring the node package, obviously it's totally fine to follow that repo's version. 
Hope the move went well and work gets better! 😃
